src/coinbase/staking_operation.ts

```diff
@@ -36,14 +37,77 @@ export class StakingOperation {
   }
 
   /**
-   * Get the staking operation ID.
+   * Returns the Staking Operation ID.
    *
-   * @returns The unique ID of the staking operation.
+   * @returns The Staking Operation ID.
    */
   public getID(): string {
     return this.model.id;
   }
 
+  /**
+   * Returns the Wallet ID if it exists.
+   *
+   * @returns The Wallet ID.
+   */
+  public getWalletID(): string | undefined {
+    return this.model.wallet_id;
+  }
+
+  /**
+   * Returns the Address ID.
+   *
+   * @returns The Address ID.
+   */
+  public getAddressID(): string {
+    return this.model.address_id;
+  }
+
+  /**
+   * Returns the Status of the StakingOperation based on the internal
+   * StakingOperation model.
+   *
+   * @returns The Status of the StakingOperation.
+   */
+  public getStatus(): StakingOperationStatus | undefined {
+    switch (this.model.status) {
+      case StakingOperationStatusEnum.Initialized:
+        return StakingOperationStatus.INITIALIZED;
+      case StakingOperationStatusEnum.Pending:
+        return StakingOperationStatus.PENDING;
+      case StakingOperationStatusEnum.Complete:
+        return StakingOperationStatus.COMPLETE;
+      case StakingOperationStatusEnum.Failed:
+        return StakingOperationStatus.FAILED;
+      default:
+        return StakingOperationStatus.UNSPECIFIED;
+    }
+  }
+
+  /**
+   * Reloads the StakingOperation model with the latest data from the server.
+   * If the StakingOperation object was created by an ExternalAddress then it will
+   * not have a wallet ID.
+   *
+   * @throws {APIError} if the API request to get the StakingOperation fails.
+   * @throws {Error} if this function is called on a StakingOperation without a wallet ID.
+   */
+  public async reload(): Promise<void> {
+    if (this.getWalletID() === undefined) {
+      throw new Error("cannot reload staking operation without a wallet ID.");
+    }
+    const result = await Coinbase.apiClients.stake!.getStakingOperation(
+      this.getWalletID()!,
+      this.getAddressID(),
+      this.getID(),
+    );
+    this.model = result?.data;
+    this.transactions = [];
```

Comment:
should we empty the transactions and then append to only when `result?.data.transactions` is non-empty to be safe ?

Reply:
Are you thinking about the scenario where we might get back an empty response and end up wiping the transactions mistakenly?

Reply:
Yes